Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpTransformBackend.cpp

```diff
@@ -42,20 +42,34 @@ void LibWebRTCRtpTransformBackend::clearTransformableFrameCallback()
     setInputCallback({ });
 }
 
-void LibWebRTCRtpTransformBackend::setOutputCallback(rtc::scoped_refptr<webrtc::TransformedFrameCallback>&& callback)
+void LibWebRTCRtpTransformBackend::addOutputCallback(rtc::scoped_refptr<webrtc::TransformedFrameCallback>&& callback, uint32_t ssrc)
 {
-    Locker locker { m_outputCallbackLock };
-    m_outputCallback = WTFMove(callback);
+    Locker locker { m_outputCallbacksLock };
+    m_outputCallbacks.insert_or_assign(ssrc, WTFMove(callback));
 }
 
-void LibWebRTCRtpTransformBackend::processTransformedFrame(RTCRtpTransformableFrame& frame)
+void LibWebRTCRtpTransformBackend::removeOutputCallback(uint32_t ssrc)
 {
-    auto rtcFrame = static_cast<LibWebRTCRtpTransformableFrame&>(frame).takeRTCFrame();
-    if (!rtcFrame)
+    Locker locker { m_outputCallbacksLock };
+    m_outputCallbacks.erase(ssrc);
+}
+
+void LibWebRTCRtpTransformBackend::sendFrameToOutput(std::unique_ptr<webrtc::TransformableFrameInterface>&& rtcFrame)
+{
+    Locker locker { m_outputCallbacksLock };
+    if (m_outputCallbacks.size() == 1) {
+        m_outputCallbacks.begin()->second->OnTransformedFrame(WTFMove(rtcFrame));
         return;
-    Locker locker { m_outputCallbackLock };
-    if (m_outputCallback)
-        m_outputCallback->OnTransformedFrame(WTFMove(rtcFrame));
+    }
+    auto iterator = m_outputCallbacks.find(rtcFrame->GetSsrc());
```

Comment:
I understand the optimisation here, but the speed difference in looking up a single map may be just the same, I wouldn't have the exception if the table as a single object. 

Or have an assert that rtcFrame->GetSsrc() is the one in the map

Reply:
This 0 check is due to libwebrtc current API which sets the callback with no SSRCs in case of no simulcast (hence ssrc = 0 in our code path).
The plan is for libwebrtc to remove the API that sets the callback without ssrcs, at which point we should only rely on checking the map like you propose.